src/Model/Endpoint.php

```diff
@@ -133,35 +133,35 @@ class Endpoint implements RepositoryInterface, EventListenerInterface, EventDisp
     public function __construct(array $config = [])
     {
         if (!empty($config['alias'])) {
-            $this->alias($config['alias']);
+            $this->setAlias($config['alias']);
         }
         if (!empty($config['connection'])) {
-            $this->connection($config['connection']);
+            $this->setConnection($config['connection']);
         }
         if (!empty($config['displayField'])) {
-            $this->displayField($config['displayField']);
+            $this->setDisplayField($config['displayField']);
         }
-        if (!empty($config['endpoint'])) {
-            $this->endpoint($config['endpoint']);
+        if (!empty($config['inflect'])) {
+            $this->setInflectionMethod($config['inflect']);
+        }
+        if (!empty($config['name'])) {
```

Comment:
To maintain BC you should also check for old name `endpoint` and if provided set `name` using that.